neqo-transport/src/packet.rs

```diff
@@ -405,26 +417,32 @@ impl<'a> PublicPacket<'a> {
                     scid: Some(scid),
                     token: &[],
                     header_len: decoder.offset(),
+                    quic_version: None,
                     data,
                 },
                 &[],
             ));
         }
 
         // Check that this is a long header from this version.
-        if version != QUIC_VERSION {
-            return Ok((
-                Self {
-                    packet_type: PacketType::OtherVersion,
-                    dcid,
-                    scid: Some(scid),
-                    token: &[],
-                    header_len: decoder.offset(),
-                    data,
-                },
-                &[],
-            ));
-        }
+        let quic_version = match QuicVersion::try_from(version) {
```

Comment:
This is probably a case for `if let Some(v) = QuicVersion::try_from(version) { v } else { return Ok((...)); }`

Reply:
seems roughly the same but sure, ok.